firebase-firestore/src/main/java/com/google/firebase/firestore/core/Transaction.java

```diff
@@ -133,13 +140,14 @@ private Precondition precondition(DocumentKey key) {
    * Returns the precondition for a document if the operation is an update, based on the provided
    * UpdateOptions.
    */
-  private Precondition preconditionForUpdate(DocumentKey key) {
+  private Precondition preconditionForUpdate(DocumentKey key) throws FirebaseFirestoreException {
     @Nullable SnapshotVersion version = this.readVersions.get(key);
     if (version != null && version.equals(SnapshotVersion.NONE)) {
       // The document to update doesn't exist, so fail the transaction.
```

Comment:
It's worth expanding on this comment because we were both confused about why we needed to do this. We need to check this locally because we can't express the constraint to the backend.

> The document to update doesn't exist, so fail the transaction.
>
> This has to be validated locally because you can't send a precondition that a document does not exist without changing the semantics of the backend write to be an insert. This is the reverse of what we want, since we want to assert that the document doesn't exist but then send the update and have it fail. Since we can't express that to the backend, we have to validate locally.
>
> Note: this can change once we can send separate verify writes in the transaction.